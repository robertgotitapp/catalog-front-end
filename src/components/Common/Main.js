import React, { Component } from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import NavBar from './NavBar';
import SignInPage from '../SignInPage/SignInPage';
import SignUpPage from '../SignUpPage/SignUpPage';
import Home from '../HomePage/Home';
import AddCategory from '../AddCategoryPage/AddCategory';
import AddItem from '../AddItemPage/AddItem';
import ItemDetail from '../ItemDetailPage/ItemDetail';
import UpdateItem from '../UpdateItemPage/UpdateItem';
import { loadCurrentUserData } from '../../actions/users';
import { getCategories } from '../../actions/categories';
import { PaginationConfig } from '../../utils/const';

class Main extends Component {
  componentDidMount() {
    this.props.loadCurrentUserData();
    this.props.getCategories(0, PaginationConfig.CATEGORIES_PER_PAGE);
  }

  render() {
    return (
      <div>
        <NavBar history={this.props.history} />
        { this.props.currentLoggedId
          ? (
            <div>
              <Route exact path='/newcategory' component={AddCategory} />
              <Route exact path='/newitem' component={AddItem} />
              <Route exact path='/items/:id/update' component={UpdateItem} />
            </div>
          )
          : (
            <div>
              <Route exact path='/signin' component={SignInPage} />
              <Route exact path='/signup' component={SignUpPage} />
            </div>
          )}
        <Route exact path='/items/:id' component={ItemDetail} />
        <Route exact path='/categories/:categoryId/items/:pageNumber' component={Home} />
        {/* <Redirect from='*' to='/' /> */}
      </div>
    );
  }
}

const mapDispatchToProps = {
  loadCurrentUserData,
  getCategories,
};

function mapStateToProps({ users }) {
  return {
    currentLoggedId: users.userId,
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Main);
