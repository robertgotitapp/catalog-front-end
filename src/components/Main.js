import React, { Component } from 'react';
import { Route, Redirect } from 'react-router-dom';
import NavBar from './NavBar';
import SignInPage from './SignInPage';
import SignUpPage from './SignUpPage';
import Home from './Home';
import AddCategory from './AddCategory';
import AddItem from './AddItem';
import ItemDetail from './ItemDetail';
import UpdateItem from './UpdateItem';

class Main extends Component {
  render() {
    return (
      <div>
        <NavBar />
        <Route exact path='/signin' component={SignInPage} />
        <Route exact path='/signup' component={SignUpPage} />
        <Route exact path='/newcategory' component={AddCategory} />
        <Route exact path='/newitem' component={AddItem} />
        <Route exact path='/items/:id' component={ItemDetail} />
        <Route exact path='/items/:id/update' component={UpdateItem} />
        <Route exact path='/' component={Home} />
        <Redirect from='*' to='/' />
      </div>
    );
  }
}

export default Main;
