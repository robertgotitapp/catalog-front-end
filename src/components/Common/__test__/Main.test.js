import React from 'react';
import { shallow } from 'enzyme';
import { Main } from '../Main';

describe('components/Home', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(<Main {...props} />);
  };

  it('should render correctly when logged in', () => {
    props = {
      loadCurrentUserData: jest.fn(),
      getCategories: jest.fn(),
      currentLoggedId: 1,
    };
    setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly when logged out', () => {
    props = {
      loadCurrentUserData: jest.fn(),
      getCategories: jest.fn(),
      currentLoggedId: 0,
    };
    setup();
    expect(wrapper).toMatchSnapshot();
  });
});
