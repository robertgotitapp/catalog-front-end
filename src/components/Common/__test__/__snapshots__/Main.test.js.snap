// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Home should render correctly when logged in 1`] = `
<div>
  <Connect(NavBar) />
  <div>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(AddCategory)",
          "type": [Function],
        }
      }
      exact={true}
      path="/newcategory"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(AddItem)",
          "type": [Function],
        }
      }
      exact={true}
      path="/newitem"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(UpdateItem)",
          "type": [Function],
        }
      }
      exact={true}
      path="/items/:id/update"
    />
  </div>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(ItemDetail)",
        "type": [Function],
      }
    }
    exact={true}
    path="/items/:id"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Home)",
        "type": [Function],
      }
    }
    exact={true}
    path="/categories/:categoryId/items/:pageNumber"
  />
</div>
`;

exports[`components/Home should render correctly when logged out 1`] = `
<div>
  <Connect(NavBar) />
  <div>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(SignInPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/signin"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(SignUpPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/signup"
    />
  </div>
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(ItemDetail)",
        "type": [Function],
      }
    }
    exact={true}
    path="/items/:id"
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(Home)",
        "type": [Function],
      }
    }
    exact={true}
    path="/categories/:categoryId/items/:pageNumber"
  />
</div>
`;
